packages/native-stack/src/types.tsx

```diff
@@ -532,17 +547,32 @@ export type NativeStackNavigationOptions = {
   /**
    * Describes heights where a sheet can rest.
    * Works only when `presentation` is set to `formSheet`.
-   * Defaults to `large`.
    *
-   * Available values:
+   * Heights should be described as fraction (a number from `[0, 1]` interval) of screen height / maximum detent height.
+   * There is also possibility to specify `fitToContents` literal, which intents to set the sheet height
+   * to the height of its contents.
    *
-   * - `large` - only large detent level will be allowed
-   * - `medium` - only medium detent level will be allowed
-   * - `all` - all detent levels will be allowed
+   * Please note that the array **must** be sorted in ascending order.
    *
-   * @platform ios
+   * There are also legacy & **deprecated** options available:
+   *
+   * * 'medium' - corresponds to `[0.5]` detent value, around half of the screen height,
+   * * 'large' - corresponds to `[1.0]` detent value, maximum height,
+   * * 'all' - corresponds to `[0.5, 1.0]` value, the name is deceiving due to compatibility reasons.
+   *
+   * Defaults to `[1.0]` literal.
    */
-  sheetAllowedDetents?: SheetDetentTypes;
+  sheetAllowedDetents?: number[] | 'fitToContents' | 'medium' | 'large' | 'all';
```

Comment:
Few points to notice here:

1. I've brought back the old-style-API values support, these now are resolved internally by `react-native-screens` (whole compatibility logic is hidden from native-stack code). This was done to mitigate the breaking changes. I've marked them as deprecated and have intention of removing them in the closer unspecified future.
2. I've moved from using `ScreenProps['sheetAllowedDetents']` etc. for a few props, to just manually specifying the type. I guess using `ScreenProps['propertyName']` has its own advantages, but I was frustrated with typescript errors throughout the native-stack code, since it does not see the changes from unpublished screens versions. **Let me know whether thats fine or not** in review, please.